centreon/src/Core/Media/Domain/Model/Media.php

```diff
@@ -53,60 +53,32 @@ public function __construct(
         $this->filename = str_replace(' ', '_', $this->filename);
         $this->directory = str_replace(' ', '', $this->directory);
         Assertion::notEmptyString($this->directory, "{$this->className}::directory");
-        Assertion::regex($this->directory, '/^[a-zA-Z0-9_-]+$/', "{$this->className}::directory");
-        if ($this->data !== null) {
-            $this->data = trim($this->data);
-            Assertion::notEmptyString($this->data, "{$this->className}::data");
+        if ($this->comment !== null) {
+            $this->comment = trim($this->comment);
         }
+        Assertion::regex($this->directory, '/^[a-zA-Z0-9_-]+$/', "{$this->className}::directory");
     }
 
     public function getId(): int
     {
         return $this->id;
     }
 
-    public function setFilename(string $filename): void
```

Comment:
Is this class could extends NewMedia ? So you could remove all the getters from this class too

Reply:
Impossible due to signature difference on 'hash' method